### PR TITLE
Added All Graph Algorithms

### LINKED ISSUE #627: [Feature Request]:  Add Graph Algorithms
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

To add Graph Algorithms in GFG Documentation:

1. Dijkstra's Algorithm: Finds the shortest path between nodes in a graph with non-negative edge weights.
2. Bellman-Ford Algorithm: Computes shortest paths from a single source node to all other nodes, even with negative weights.
3. Floyd-Warshall Algorithm: Finds shortest paths between all pairs of nodes.
4. Kruskal’s and Prim’s Algorithms: Used for finding the minimum spanning tree in a graph.

### Use Case

Widely used in Solving GRAPH Hard problems.

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by MuraliDharan7**
## Related Issue
- Closes #627 

## Description
Added the Graph algorithms in Algorithms section.
1.Dijkstra's Algorithm: Finds the shortest path between nodes in a graph with non-negative edge weights.
2. Bellman-Ford Algorithm: Computes shortest paths from a single source node to all other nodes, even with negative weights.
3. Floyd-Warshall Algorithm: Finds shortest paths between all pairs of nodes.
4. Kruskal’s and Prim’s Algorithms: Used for finding the minimum spanning tree in a graph.

## Type of PR

- [X] Documentation update

## Screenshots / Videos (if applicable)
image

image

## Checklist
- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

**Comment by MuraliDharan7** [comment]
@Ajay-Dhangar  Can you guide me why does this error occurs..

**Comment by Ajay-Dhangar** [comment]
> @Ajay-Dhangar Can you guide me why does this error occurs..

yah sure

File: dsa-solutions/index.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ In this section, you will learn about the solutions of the problems of Data Stru
 - [Backtracking](#): Backtracking is a technique used to solve problems by making a series of choices and backtracking when a choice leads to a dead end.
 - [Divide and Conquer](#): Divide and Conquer is a technique used to solve problems by breaking them down into smaller subproblems, solving the subproblems, and combining the solutions to solve the original problem.
 - [Bit Manipulation](#): Bit Manipulation is the act of algorithmically manipulating bits or binary digits.
-- [Graph Algorithms](#): Graph Algorithms are algorithms that operate on graphs to solve problems.
+- [Graph Algorithms](../dsa/Algorithms/Graph%20Algorithms): Graph Algorithms are algorithms that operate on graphs to solve problems.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/index.md:32]: replace `Graph%20Algorithms` to `graph-algorithms`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions and resolve issues / errors

**Comment by MuraliDharan7** [comment]
> update your code based on suggestions and resolve issues / errors

Still It has the error, Even after the changes are done

**Comment by Ajay-Dhangar** [comment]
@MuraliDharan7  please use the correct link because the path link is incorrect.

File: dsa-solutions/index.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ In this section, you will learn about the solutions of the problems of Data Stru
 - [Backtracking](#): Backtracking is a technique used to solve problems by making a series of choices and backtracking when a choice leads to a dead end.
 - [Divide and Conquer](#): Divide and Conquer is a technique used to solve problems by breaking them down into smaller subproblems, solving the subproblems, and combining the solutions to solve the original problem.
 - [Bit Manipulation](#): Bit Manipulation is the act of algorithmically manipulating bits or binary digits.
-- [Graph Algorithms](#): Graph Algorithms are algorithms that operate on graphs to solve problems.
+- [Graph Algorithms](../dsa/Algorithms/graph-algorithms): Graph Algorithms are algorithms that operate on graphs to solve problems.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/index.md:32]: replace `../dsa/Algorithms/graph-algorithms` to `/dsa/Algorithms/graph-algorithms`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by MuraliDharan7** [comment]
Since this is keep on getting errors. Can I request new PR for this issue?

**Comment by MuraliDharan7** [comment]
> update your code based on suggestions

Changes are done.
Since this is keep on getting errors. Can I request new PR for this issue?

File: dsa-solutions/index.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ In this section, you will learn about the solutions of the problems of Data Stru
 - [Backtracking](#): Backtracking is a technique used to solve problems by making a series of choices and backtracking when a choice leads to a dead end.
 - [Divide and Conquer](#): Divide and Conquer is a technique used to solve problems by breaking them down into smaller subproblems, solving the subproblems, and combining the solutions to solve the original problem.
 - [Bit Manipulation](#): Bit Manipulation is the act of algorithmically manipulating bits or binary digits.
-- [Graph Algorithms](#): Graph Algorithms are algorithms that operate on graphs to solve problems.
+- [Graph Algorithms](/dsa/Algorithms/graph-algorithms): Graph Algorithms are algorithms that operate on graphs to solve problems.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/index.md:32]: You can replace `/dsa/Algorithms/graph-algorithms` with `#`. I will update it. I believe `/dsa/category/graph-algorithms` is working, so please try that instead. Additionally, ensure you are running the project on your local system, as this is a better way to solve errors.

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Now, adjust the code according to this approach. I trust you have resolved the error.

**Comment by Ajay-Dhangar** [comment]
@MuraliDharan7 complete your task very soon

**Comment by MuraliDharan7** [comment]
> Now, adjust the code according to this approach. I trust you can solve the error.

All have passed. Thank you for your guidance.

**Review by Ajay-Dhangar — APPROVED** [review]
done

**Pull request merged**